src/App.tsx

```diff
@@ -1,24 +1,209 @@
+    todos.filter(todo => !todo.completed).length
+  ), [todos]);
+
+  const addTodo = useCallback(async (title: string) => {
+    if (!title.trim()) {
+      setHasError(ErrorMessage.EMPTY_TITLE);
+
+      return;
+    }
+
+    const newTodo = {
+      title,
+      completed: false,
+      userId: USER_ID,
+    };
+
+    try {
+      setIsDisabled(true);
+      setTempTodo({ id: 0, ...newTodo });
+
+      const todo = await createTodo(newTodo as Todo);
+
+      setTodos(prevState => [...prevState, todo]);
+    } catch {
+      setHasError(ErrorMessage.ADD);
+    } finally {
+      setTempTodo(null);
+      setIsDisabled(false);
+    }
+  }, [todos]);
+
+  const updateTodo = useCallback(async (
+    todoId: number,
+    updatedDate: Partial<Todo>,
+  ) => {
+    if (loadTodoId.includes(todoId)) {
+      return;
+    }
+
+    setLoadTodoId(prevState => [...prevState, todoId]);
+    setIsDisabled(true);
+
+    try {
+      const updatedTodo = await patchTodo(todoId, updatedDate);
+
+      setTodos(prevState => prevState.map(todo => (
+        todo.id === todoId
+          ? updatedTodo
+          : todo)));
+    } catch {
+      setHasError(ErrorMessage.UPDATE);
+    } finally {
+      setIsDisabled(false);
+      setLoadTodoId([0]);
+    }
+  }, [loadTodoId]);
+
+  const deleteTodo = useCallback(async (taskId: number) => {
+    setLoadTodoId(prevState => [...prevState, taskId]);
+
+    try {
+      await removeTodo(taskId);
+
+      setTodos(prevTodos => (
+        prevTodos.filter(({ id }) => id !== taskId)
+      ));
+    } catch {
+      setHasError(ErrorMessage.DELETE);
+    } finally {
+      setLoadTodoId([0]);
+    }
+  }, []);
+
+  const handleClearCompleted = () => {
+    const filteredTodos = todos.filter(todo => !todo.completed);
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    const promises = completedTodos.map(todo => {
+      setLoadTodoId(prevState => [...prevState, todo.id]);
+
+      return removeTodo(todo.id);
+    });
+
+    Promise.all(promises).then(() => {
+      setTodos(filteredTodos);
+      setLoadTodoId([]);
+    });
+  };
+
+  const activeTodos = getVisibleTodos(todos, TodoStatus.ACTIVE);
+  const completedTodos = getVisibleTodos(todos, TodoStatus.COMPLETED);
+
+  const changeStatusForAll = useCallback(async () => {
+    if (loadTodoId.length === 0) {
+      return;
+    }
+
+    await Promise.all(activeTodos.map(({ id }) => (
+      updateTodo(id, { completed: true }))));
+
+    if (!activeTodos.length) {
+      await Promise.all(
+        completedTodos.map(({ id }) => (
+          updateTodo(id, { completed: false }))),
+      );
+    }
+  }, [completedTodos, activeTodos]);
+
+  const onCloseError = () => setHasError(ErrorMessage.NONE);
+
+  const visibleTodos = getVisibleTodos(todos, todoStatus);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((res) => {
+        setTodos(res);
+      })
+      .catch(() => {
+        setHasError(ErrorMessage.LOAD);
+      });
```

Comment:
I'd recommend to make async operations in one style (using async/await as you did above)
You can create a function inside this useEffect (const fetchData = async ...). And then invoke this function. Or even you can use `IIFE`